src/vs/workbench/parts/preferences/browser/preferencesRenderers.ts

```diff
@@ -297,20 +307,31 @@ export class DefaultSettingsRenderer extends Disposable implements IPreferencesR
 
 	public filterPreferences(filterResult: IFilterResult): void {
 		this.filterResult = filterResult;
-		if (!filterResult) {
+		if (filterResult) {
+			if (filterResult.remoteResult) {
```

Comment:
I would not like to leak the fact that if results are remote or not. Both the results provider could return results using common API. The editor and renderers just depends on this results API.

Reply:
As above, there are differences in rendering, but the "remote" aspect can be hidden.